app/src/main/java/fr/free/nrw/commons/settings/SettingsFragment.java

```diff
@@ -169,15 +173,17 @@ private void prepareAppLanguages(final String keyListPreference) {
             Objects.requireNonNull(getContext()));
         languageNamesList = appLanguageLookUpTable.getLocalizedNames();
         languageCodesList = appLanguageLookUpTable.getCodes();
-        List<String> languageNameWithCodeList = new ArrayList<>();
+        final SortedMap<String, String> languageMap = new TreeMap<>(String.CASE_INSENSITIVE_ORDER);
```

Comment:
I guess this is not related to #4590? :-)

Reply:
This code is related to issue #4502 and PR #4623 . 

@jmora13 For solving each issue create a different branch from **master** to avoid this kind of overlapping.